### PR TITLE
Add support for Winkel Tripel projection

### PR DESCRIPTION
**Pull request by chriddyp**
the winkel tripel projection minimizes distortions in [shape, flexion, and skewness](http://usersguidetotheuniverse.com/index.php/2011/03/03/whats-the-best-map-projection/). It's also the [recommended projection from the cartographers at National Geographic](http://maps.nationalgeographic.com/maps/print-collection/world-map-bright.html). 

I'd like to make this projection available as a default in Plotly 2.0 :)

### CODE REVIEW

**Comment by etpinard** [comment]
:+1: That's a dope projection: 

image

http://bl.ocks.org/mbostock/3682676

In order to get it in plotly.js, we'll need to update [`src/plots/geo/projections.js`](https://github.com/plotly/plotly.js/blob/master/src/plots/geo/projections.js) using my [`d3-geo-projection-picker`](https://github.com/etpinard/d3-geo-projection-picker). By the way, I don't want to make `d3-geo-projection-picker` a plotly.js dependency because of its nasty `postinstall` hook.

@chriddyp feel free to give it a shot or I'll can get to it at some point this week.

**Comment by etpinard** [comment]
@chriddyp hope you like it:

image

**Comment by etpinard** [comment]
@mdtusz would you mind reviewing this. Thanks

File: src/plots/geo/projections.js [hunk 1]
```diff
@@ -405,6 +406,39 @@ function addProjectionToD3() {
   (d3.geo.sinusoidal = function() {
     return projection(sinusoidal);
   }).raw = sinusoidal;
+  function aitoff(λ, φ) {
```
Comment (mdtusz) [inline src/plots/geo/projections.js]: Uhhh...

File: src/plots/geo/projections.js [hunk 2]
```diff
@@ -405,6 +406,39 @@ function addProjectionToD3() {
...
+    return projection(aitoff);
+  }).raw = aitoff;
+  function winkel3(λ, φ) {
+    var coordinates = aitoff(λ, φ);
+    return [ (coordinates[0] + λ / halfπ) / 2, (coordinates[1] + φ) / 2 ];
+  }
+  winkel3.invert = function(x, y) {
+    var λ = x, φ = y, i = 25;
+    do {
+      var cosφ = Math.cos(φ), sinφ = Math.sin(φ), sin_2φ = Math.sin(2 * φ), sin2φ = sinφ * sinφ, cos2φ = cosφ * cosφ, sinλ = Math.sin(λ), cosλ_2 = Math.cos(λ / 2), sinλ_2 = Math.sin(λ / 2), sin2λ_2 = sinλ_2 * sinλ_2, C = 1 - cos2φ * cosλ_2 * cosλ_2, E = C ? acos(cosφ * cosλ_2) * Math.sqrt(F = 1 / C) : F = 0, F, fx = .5 * (2 * E * cosφ * sinλ_2 + λ / halfπ) - x, fy = .5 * (E * sinφ + φ) - y, δxδλ = .5 * F * (cos2φ * sin2λ_2 + E * cosφ * cosλ_2 * sin2φ) + .5 / halfπ, δxδφ = F * (sinλ * sin_2φ / 4 - E * sinφ * sinλ_2), δyδλ = .125 * F * (sin_2φ * sinλ_2 - E * sinφ * cos2φ * sinλ), δyδφ = .5 * F * (sin2φ * cosλ_2 + E * sin2λ_2 * cosφ) + .5, denominator = δxδφ * δyδλ - δyδφ * δxδλ, δλ = (fy * δxδφ - fx * δyδφ) / denominator, δφ = (fx * δyδλ - fy * δxδλ) / denominator;
```
Comment (mdtusz) [inline src/plots/geo/projections.js]: Wat even.

**Pull request merged**